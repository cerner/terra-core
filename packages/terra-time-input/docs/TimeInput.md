| Prop Name | Type | Is Required | Default Value | Description |
|-|-|-|-|-|
| inputAttributes| `object`| optional| `undefined`| Custom input attributes to apply to the time input.|
| onChange| `func`| optional| `null`| A callback function to execute when either the hour or minute has been changed. The first parameter is the event. The second parameter is the changed time value.|
| value| `string`| optional| `undefined`| An ISO 8601 string representation of the time value in the input.|