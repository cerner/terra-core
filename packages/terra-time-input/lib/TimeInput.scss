@import '~terra-mixins';

.terra-TimeInput {
  display: inline-block;
  min-width: 12em; 
}

.terra-TimeInput-input {
  @include terra-border-radius-start(0.286em);
  @include terra-border-radius-end(0.286em);
  border: 0.071em solid $terra-grey-40;
  box-shadow: inset 0 0.143em 0.143em $terra-grey-20;
  font-size: 1em;
  line-height: 1.143;
  margin: 0.357em 0;
  padding: 0.429em 0.714em;
  vertical-align: middle;

  &:focus {
    border-color: $terra-blue-60;
    outline: none;
  }
}
