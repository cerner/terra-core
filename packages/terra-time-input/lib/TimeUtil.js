'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _moment = require('moment');

var _moment2 = _interopRequireDefault(_moment);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

var TimeUtil = function () {
  function TimeUtil() {
    _classCallCheck(this, TimeUtil);
  }

  _createClass(TimeUtil, null, [{
    key: 'formattedTime',
    value: function formattedTime(time, format) {
      if (time && format) {
        var momentTime = (0, _moment2.default)(time, format, true);
        if (momentTime.isValid()) {
          return momentTime.format(format);
        }
      }

      return undefined;
    }
  }]);

  return TimeUtil;
}();

exports.default = TimeUtil;