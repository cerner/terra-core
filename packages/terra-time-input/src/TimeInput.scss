// stylelint-disable declaration-property-value-blacklist
/* stylelint-disable selector-class-pattern  */
:local {
  .time-input {
    background-color: transparent;
    border: #dedfe0 solid 1px;
    border-bottom-left-radius: 0.25em;
    border-bottom-right-radius: 0.25em;
    border-top-left-radius: 0.25em;
    border-top-right-radius: 0.25em;
    display: inline-block;
    margin: 0;

    &.is-focused {
      border-color: #26a2e5;
      outline: none;
    }
  }

  .time-input-hour,
  .time-input-minute {
    border: none;
    display: inline-block;
    margin: 0;
    min-width: 2.1em;
    padding: 0.179em 0;
    text-align: center;
    width: 2.1em;

    &:focus {
      border-color: none;
      outline: none;
    }
  }
}
/* stylelint-enable */
