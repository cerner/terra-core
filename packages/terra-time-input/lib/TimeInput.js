'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _moment = require('moment');

var _moment2 = _interopRequireDefault(_moment);

var _reactTextMask = require('react-text-mask');

var _reactTextMask2 = _interopRequireDefault(_reactTextMask);

require('terra-base/lib/baseStyles');

var _TimePipe = require('./TimePipe');

var _TimePipe2 = _interopRequireDefault(_TimePipe);

var _TimeUtil = require('./TimeUtil');

var _TimeUtil2 = _interopRequireDefault(_TimeUtil);

require('./TimeInput.scss');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var propTypes = {
  /**
   * An ISO 8601 string representation of the default time.
   */
  defaultValue: _react.PropTypes.string,
  /**
   * Custom input attributes to apply to the time input.
   */
  inputAttributes: _react.PropTypes.object,
  /**
  * A callback function to execute when a time value is entered.
  */
  onChange: _react.PropTypes.func,
  /**
   * An ISO 8601 string representation of the time value in the input.
   */
  value: _react.PropTypes.string
};

var defaultProps = {
  inputAttributes: undefined,
  defaultValue: undefined,
  onChange: null,
  value: undefined
};

var TimeInput = function (_React$Component) {
  _inherits(TimeInput, _React$Component);

  _createClass(TimeInput, null, [{
    key: 'userLocale',
    value: function userLocale() {
      // TODO: Get the locale from i18nProvider
      return 'en-US';
    }
  }, {
    key: 'timeFormat',
    value: function timeFormat() {
      // TODO: Get the time format translation.
      return 'HH:mm';
    }
  }]);

  function TimeInput(props) {
    _classCallCheck(this, TimeInput);

    var _this = _possibleConstructorReturn(this, (TimeInput.__proto__ || Object.getPrototypeOf(TimeInput)).call(this, props));

    _this.state = {
      defaultValue: _TimeUtil2.default.formattedTime(props.defaultValue, TimeInput.timeFormat()),
      value: _TimeUtil2.default.formattedTime(props.value, TimeInput.timeFormat())
    };

    _this.handleChange = _this.handleChange.bind(_this);
    _this.handleInputKeyDown = _this.handleInputKeyDown.bind(_this);
    return _this;
  }

  _createClass(TimeInput, [{
    key: 'handleChange',
    value: function handleChange(event) {
      if (event.target.value === this.state.value) {
        return;
      }

      this.setState({
        value: event.target.value
      });

      if (this.props.onChange) {
        this.props.onChange(event.target.value, event);
      }
    }
  }, {
    key: 'handleInputKeyDown',
    value: function handleInputKeyDown(event) {
      var momentTime = (0, _moment2.default)(this.state.value, TimeInput.timeFormat(), true);

      if (!momentTime.isValid()) {
        return;
      }

      var updateTime = false;

      if (event.key === 'ArrowUp') {
        momentTime = momentTime.add(1, 'minutes');
        updateTime = true;
      } else if (event.key === 'ArrowDown') {
        momentTime = momentTime.subtract(1, 'minutes');
        updateTime = true;
      }

      if (updateTime) {
        var incrementedTime = momentTime.format(TimeInput.timeFormat());
        this.setState({ value: incrementedTime });
      }
    }
  }, {
    key: 'render',
    value: function render() {
      var _props = this.props,
          inputAttributes = _props.inputAttributes,
          defaultValue = _props.defaultValue,
          onChange = _props.onChange,
          value = _props.value,
          customProps = _objectWithoutProperties(_props, ['inputAttributes', 'defaultValue', 'onChange', 'value']);

      var attributes = _extends({}, customProps, inputAttributes);

      return _react2.default.createElement(
        'div',
        { className: 'terra-TimeInput' },
        _react2.default.createElement(_reactTextMask2.default, _extends({}, attributes, {
          className: 'terra-TimeInput-input',
          type: 'text',
          value: this.state.value,
          defaultValue: this.state.defaultValue,
          onChange: this.handleChange,
          placeholder: TimeInput.timeFormat(),
          mask: [/[0-2]/, /[0-9]/, ':', /[0-5]/, /[0-9]/],
          keepCharPositions: true,
          placeholderChar: ' ',
          pipe: (0, _TimePipe2.default)(TimeInput.timeFormat()),
          onKeyDown: this.handleInputKeyDown
        }))
      );
    }
  }]);

  return TimeInput;
}(_react2.default.Component);

TimeInput.propTypes = propTypes;
TimeInput.defaultProps = defaultProps;

exports.default = TimeInput;