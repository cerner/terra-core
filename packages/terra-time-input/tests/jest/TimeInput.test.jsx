/* globals spyOn jest */

import React from 'react';
import intlContexts from './intl-context-setup';
import TimeInput from '../../src/TimeInput';
import TimeUtil from '../../src/TimeUtil';

const mockEvent = {
  preventDefault: jest.fn(),
};

const handleOnChange = () => {
};

const handleOnBlur = () => {
};

it('should render a default time input', () => {
  const timeInput = <TimeInput name="time-input" />;
  const wrapper = shallow(timeInput, intlContexts.shallowContext);
  expect(wrapper).toMatchSnapshot();
});

it('should render a time input with custom attributes', () => {
  const timeInput = <TimeInput name="time-input" inputAttributes={{ id: 'terra-time-input' }} />;
  const wrapper = shallow(timeInput, intlContexts.shallowContext);
  expect(wrapper).toMatchSnapshot();
});

it('should render a time input with a default time', () => {
  const timeInput = <TimeInput name="time-input" value={'10:45'} />;
  const wrapper = shallow(timeInput, intlContexts.shallowContext);
  expect(wrapper).toMatchSnapshot();
});

it('should render a time input with onChange', () => {
  const timeInput = <TimeInput name="time-input" onChange={handleOnChange} />;
  const wrapper = shallow(timeInput, intlContexts.shallowContext);
  expect(wrapper).toMatchSnapshot();
});

it('should render a time input with onBlur', () => {
  const timeInput = <TimeInput name="time-input" onBlur={handleOnBlur} />;
  const wrapper = shallow(timeInput, intlContexts.shallowContext);
  expect(wrapper).toMatchSnapshot();
});

it('should render a time input with onInputFocus', () => {
  const timeInput = <TimeInput name="time-input" onBlur={() => {}} />;
  const wrapper = shallow(timeInput, intlContexts.shallowContext);
  expect(wrapper).toMatchSnapshot();
});

it('should render a time input with 12 hour clock set', () => {
  const timeInput = <TimeInput name="time-input" variant="12-hour" />;
  const wrapper = shallow(timeInput, intlContexts.shallowContext);
  expect(wrapper).toMatchSnapshot();
});

it('should ignore invalid times properly', () => {
  const timeInput = <TimeInput name="time-input" value="11:2" />;
  const wrapper = shallow(timeInput, intlContexts.shallowContext);
  expect(wrapper).toMatchSnapshot();

  expect(wrapper.instance().state.hour).toEqual('');
  expect(wrapper.instance().state.minute).toEqual('');
});

it('should render a 24 hour timepicker properly on mobile devices', () => {
  spyOn(window, 'matchMedia').and.returnValue({ matches: true });
  window.ontouchstart = 'true';

  const timeInput = <TimeInput name="time-input" />;
  const wrapper = render(timeInput, intlContexts.mountContext);
  expect(wrapper).toMatchSnapshot();
  delete window.ontouchstart;
});

it('should set the labels with the correct htmlFor a 24 hour timepicker properly on mobile devices', () => {
  spyOn(window, 'matchMedia').and.returnValue({ matches: true });
  window.ontouchstart = 'true';

  const timeInput = <TimeInput name="time-input" minuteAttributes={{ id: 'id-1' }} hourAttributes={{ id: 'id-2' }} />;
  const wrapper = render(timeInput, intlContexts.mountContext);
  expect(wrapper).toMatchSnapshot();
  delete window.ontouchstart;
});

it('should render a 12 hour timepicker meridiem with buttons when viewed on a mobile device', () => {
  spyOn(window, 'matchMedia').and.returnValue({ matches: true });
  window.ontouchstart = 'true';

  const timeInput = <TimeInput name="time-input" variant="12-hour" />;
  const wrapper = render(timeInput, intlContexts.mountContext);
  expect(wrapper).toMatchSnapshot();
  delete window.ontouchstart;
});

it('should render a disabled time input', () => {
  const timeInput = <TimeInput name="time-input" disabled />;
  const wrapper = shallow(timeInput, intlContexts.shallowContext);
  expect(wrapper).toMatchSnapshot();
});

it('should handle focusing on the hour input without error', () => {
  mockEvent.keyCode = TimeUtil.keyCodes.ARROWRIGHT;
  const timeInput = <TimeInput name="time-input" disabled />;
  const wrapper = mount(timeInput, intlContexts.shallowContext);
  wrapper.instance().handleHourInputKeyDown(mockEvent);
  expect(mockEvent.preventDefault).toHaveBeenCalled();
});

it('should handle focusing on the minute input without error', () => {
  mockEvent.keyCode = TimeUtil.keyCodes.ARROWLEFT;
  const timeInput = <TimeInput name="time-input" disabled />;
  const wrapper = mount(timeInput, intlContexts.shallowContext);
  wrapper.instance().setState({ hour: 2 });
  wrapper.instance().handleMinuteInputKeyDown(mockEvent);
  expect(mockEvent.preventDefault).toHaveBeenCalled();
});

it('should handle focusing on the meridiem input without error', () => {
  mockEvent.keyCode = TimeUtil.keyCodes.ARROWRIGHT;
  const timeInput = <TimeInput name="time-input" variant="12-hour" />;
  const wrapper = mount(timeInput, intlContexts.shallowContext);
  wrapper.instance().handleMinuteInputKeyDown(mockEvent);
  expect(mockEvent.preventDefault).toHaveBeenCalled();
});
