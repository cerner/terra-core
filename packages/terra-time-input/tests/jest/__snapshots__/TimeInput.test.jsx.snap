// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should ignore invalid times properly 1`] = `
<div
  className="time-input"
>
  <input
    name="time-input"
    type="hidden"
    value=""
  />
  <Input
    aria-label="Hours"
    className="time-input-hour desktop"
    disabled={false}
    isInvalid={false}
    maxLength="2"
    name="terra-time-hour-time-input"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    pattern="\\\\d*"
    placeholder="hh"
    refCallback={[Function]}
    required={false}
    size="2"
    type="text"
    value=""
  />
  <span
    className="time-spacer"
  >
    :
  </span>
  <Input
    aria-label="Minutes"
    className="time-input-minute desktop"
    disabled={false}
    isInvalid={false}
    maxLength="2"
    name="terra-time-minute-time-input"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    pattern="\\\\d*"
    placeholder="mm"
    refCallback={[Function]}
    required={false}
    size="2"
    type="text"
    value=""
  />
</div>
`;

exports[`should not have duplicate ids on the page when multiple date pickers are initialized 1`] = `
<div>
  <div
    class="mobile-time-picker"
  >
    <input
      name="time-input"
      type="hidden"
      value=""
    />
    <div
      class="time-input-group"
    >
      <input
        class="form-input time-input-hour"
        id="terra-time-hour-time-input"
        maxlength="2"
        name="terra-time-hour-time-input"
        pattern="\\\\d*"
        placeholder="hh"
        size="2"
        value=""
      />
      <label
        class="mobile-input-label"
        for="terra-time-hour-time-input"
      >
        Hours
      </label>
    </div>
    <span
      class="time-spacer"
    >
      :
    </span>
    <div
      class="time-input-group"
    >
      <input
        class="form-input time-input-minute"
        id="terra-time-minute-time-input"
        maxlength="2"
        name="terra-time-minute-time-input"
        pattern="\\\\d*"
        placeholder="mm"
        size="2"
        value=""
      />
      <label
        class="mobile-input-label"
        for="terra-time-minute-time-input"
      >
        Minutes
      </label>
    </div>
    <div
      class="button-group"
    >
      <button
        aria-disabled="false"
        aria-pressed="true"
        class="button neutral button-group-button is-active meridiem-button"
        type="button"
      >
        <span
          class="button-label text-only"
        >
          <span
            class=""
          >
            a.m.
          </span>
        </span>
      </button>
      <button
        aria-disabled="false"
        aria-pressed="false"
        class="button neutral button-group-button meridiem-button"
        type="button"
      >
        <span
          class="button-label text-only"
        >
          <span
            class=""
          >
            p.m.
          </span>
        </span>
      </button>
    </div>
  </div>
  <div
    class="mobile-time-picker"
  >
    <input
      name="time-input-2"
      type="hidden"
      value=""
    />
    <div
      class="time-input-group"
    >
      <input
        class="form-input time-input-hour"
        id="terra-time-hour-time-input-2"
        maxlength="2"
        name="terra-time-hour-time-input-2"
        pattern="\\\\d*"
        placeholder="hh"
        size="2"
        value=""
      />
      <label
        class="mobile-input-label"
        for="terra-time-hour-time-input-2"
      >
        Hours
      </label>
    </div>
    <span
      class="time-spacer"
    >
      :
    </span>
    <div
      class="time-input-group"
    >
      <input
        class="form-input time-input-minute"
        id="terra-time-minute-time-input-2"
        maxlength="2"
        name="terra-time-minute-time-input-2"
        pattern="\\\\d*"
        placeholder="mm"
        size="2"
        value=""
      />
      <label
        class="mobile-input-label"
        for="terra-time-minute-time-input-2"
      >
        Minutes
      </label>
    </div>
    <div
      class="button-group"
    >
      <button
        aria-disabled="false"
        aria-pressed="true"
        class="button neutral button-group-button is-active meridiem-button"
        type="button"
      >
        <span
          class="button-label text-only"
        >
          <span
            class=""
          >
            a.m.
          </span>
        </span>
      </button>
      <button
        aria-disabled="false"
        aria-pressed="false"
        class="button neutral button-group-button meridiem-button"
        type="button"
      >
        <span
          class="button-label text-only"
        >
          <span
            class=""
          >
            p.m.
          </span>
        </span>
      </button>
    </div>
  </div>
  <div
    class="mobile-time-picker"
  >
    <input
      name="time-input-3"
      type="hidden"
      value=""
    />
    <div
      class="time-input-group"
    >
      <input
        class="form-input time-input-hour"
        id="terra-time-hour-time-input-3"
        maxlength="2"
        name="terra-time-hour-time-input-3"
        pattern="\\\\d*"
        placeholder="hh"
        size="2"
        value=""
      />
      <label
        class="mobile-input-label"
        for="terra-time-hour-time-input-3"
      >
        Hours
      </label>
    </div>
    <span
      class="time-spacer"
    >
      :
    </span>
    <div
      class="time-input-group"
    >
      <input
        class="form-input time-input-minute"
        id="terra-time-minute-time-input-3"
        maxlength="2"
        name="terra-time-minute-time-input-3"
        pattern="\\\\d*"
        placeholder="mm"
        size="2"
        value=""
      />
      <label
        class="mobile-input-label"
        for="terra-time-minute-time-input-3"
      >
        Minutes
      </label>
    </div>
    <div
      class="button-group"
    >
      <button
        aria-disabled="false"
        aria-pressed="true"
        class="button neutral button-group-button is-active meridiem-button"
        type="button"
      >
        <span
          class="button-label text-only"
        >
          <span
            class=""
          >
            a.m.
          </span>
        </span>
      </button>
      <button
        aria-disabled="false"
        aria-pressed="false"
        class="button neutral button-group-button meridiem-button"
        type="button"
      >
        <span
          class="button-label text-only"
        >
          <span
            class=""
          >
            p.m.
          </span>
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`should render a 12 hour timepicker meridiem with buttons when viewed on a mobile device 1`] = `
<div
  class="mobile-time-picker"
>
  <input
    name="time-input"
    type="hidden"
    value=""
  />
  <div
    class="time-input-group"
  >
    <input
      class="form-input time-input-hour"
      id="terra-time-hour-time-input"
      maxlength="2"
      name="terra-time-hour-time-input"
      pattern="\\\\d*"
      placeholder="hh"
      size="2"
      value=""
    />
    <label
      class="mobile-input-label"
      for="terra-time-hour-time-input"
    >
      Hours
    </label>
  </div>
  <span
    class="time-spacer"
  >
    :
  </span>
  <div
    class="time-input-group"
  >
    <input
      class="form-input time-input-minute"
      id="terra-time-minute-time-input"
      maxlength="2"
      name="terra-time-minute-time-input"
      pattern="\\\\d*"
      placeholder="mm"
      size="2"
      value=""
    />
    <label
      class="mobile-input-label"
      for="terra-time-minute-time-input"
    >
      Minutes
    </label>
  </div>
  <div
    class="button-group"
  >
    <button
      aria-disabled="false"
      aria-pressed="true"
      class="button neutral button-group-button is-active meridiem-button"
      type="button"
    >
      <span
        class="button-label text-only"
      >
        <span
          class=""
        >
          a.m.
        </span>
      </span>
    </button>
    <button
      aria-disabled="false"
      aria-pressed="false"
      class="button neutral button-group-button meridiem-button"
      type="button"
    >
      <span
        class="button-label text-only"
      >
        <span
          class=""
        >
          p.m.
        </span>
      </span>
    </button>
  </div>
</div>
`;

exports[`should render a 24 hour timepicker properly on mobile devices 1`] = `
<div
  class="mobile-time-picker"
>
  <input
    name="time-input"
    type="hidden"
    value=""
  />
  <div
    class="time-input-group"
  >
    <input
      class="form-input time-input-hour"
      id="terra-time-hour-time-input"
      maxlength="2"
      name="terra-time-hour-time-input"
      pattern="\\\\d*"
      placeholder="hh"
      size="2"
      value=""
    />
    <label
      class="mobile-input-label"
      for="terra-time-hour-time-input"
    >
      Hours
    </label>
  </div>
  <span
    class="time-spacer"
  >
    :
  </span>
  <div
    class="time-input-group"
  >
    <input
      class="form-input time-input-minute"
      id="terra-time-minute-time-input"
      maxlength="2"
      name="terra-time-minute-time-input"
      pattern="\\\\d*"
      placeholder="mm"
      size="2"
      value=""
    />
    <label
      class="mobile-input-label"
      for="terra-time-minute-time-input"
    >
      Minutes
    </label>
  </div>
</div>
`;

exports[`should render a default time input 1`] = `
<div
  className="time-input"
>
  <input
    name="time-input"
    type="hidden"
    value=""
  />
  <Input
    aria-label="Hours"
    className="time-input-hour desktop"
    disabled={false}
    isInvalid={false}
    maxLength="2"
    name="terra-time-hour-time-input"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    pattern="\\\\d*"
    placeholder="hh"
    refCallback={[Function]}
    required={false}
    size="2"
    type="text"
    value=""
  />
  <span
    className="time-spacer"
  >
    :
  </span>
  <Input
    aria-label="Minutes"
    className="time-input-minute desktop"
    disabled={false}
    isInvalid={false}
    maxLength="2"
    name="terra-time-minute-time-input"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    pattern="\\\\d*"
    placeholder="mm"
    refCallback={[Function]}
    required={false}
    size="2"
    type="text"
    value=""
  />
</div>
`;

exports[`should render a disabled time input 1`] = `
<div
  className="disabled time-input"
>
  <input
    name="time-input"
    type="hidden"
    value=""
  />
  <Input
    aria-label="Hours"
    className="time-input-hour desktop"
    disabled={true}
    isInvalid={false}
    maxLength="2"
    name="terra-time-hour-time-input"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    pattern="\\\\d*"
    placeholder="hh"
    refCallback={[Function]}
    required={false}
    size="2"
    type="text"
    value=""
  />
  <span
    className="time-spacer"
  >
    :
  </span>
  <Input
    aria-label="Minutes"
    className="time-input-minute desktop"
    disabled={true}
    isInvalid={false}
    maxLength="2"
    name="terra-time-minute-time-input"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    pattern="\\\\d*"
    placeholder="mm"
    refCallback={[Function]}
    required={false}
    size="2"
    type="text"
    value=""
  />
</div>
`;

exports[`should render a time input with 12 hour clock set 1`] = `
<div
  className="time-input"
>
  <input
    name="time-input"
    type="hidden"
    value=""
  />
  <Input
    aria-label="Hours"
    className="time-input-hour desktop"
    disabled={false}
    isInvalid={false}
    maxLength="2"
    name="terra-time-hour-time-input"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    pattern="\\\\d*"
    placeholder="hh"
    refCallback={[Function]}
    required={false}
    size="2"
    type="text"
    value=""
  />
  <span
    className="time-spacer"
  >
    :
  </span>
  <Input
    aria-label="Minutes"
    className="time-input-minute desktop"
    disabled={false}
    isInvalid={false}
    maxLength="2"
    name="terra-time-minute-time-input"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    pattern="\\\\d*"
    placeholder="mm"
    refCallback={[Function]}
    required={false}
    size="2"
    type="text"
    value=""
  />
  <Input
    aria-label="Display Meridiem"
    aria-readonly={true}
    className="meridiem-display"
    disabled={false}
    isInvalid={false}
    key="meridiem_display"
    name={null}
    onFocus={[Function]}
    readOnly={true}
    required={false}
    size={4}
    tabIndex="-1"
    value="a.m."
  />
  <div
    key="meridiem_select_box"
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <select
      aria-label="Select Meridiem"
      className="time-input-meridiem"
      name="terra-time-meridiem-time-input"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      size="2"
      value="a.m."
    >
      <option
        key="anteMeridiem"
        value="a.m."
      >
        a.m.
      </option>
      <option
        key="postMeridiem"
        value="p.m."
      >
        p.m.
      </option>
    </select>
  </div>
</div>
`;

exports[`should render a time input with a default time 1`] = `
<div
  className="time-input"
>
  <input
    name="time-input"
    type="hidden"
    value="T10:45"
  />
  <Input
    aria-label="Hours"
    className="time-input-hour desktop"
    disabled={false}
    isInvalid={false}
    maxLength="2"
    name="terra-time-hour-time-input"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    pattern="\\\\d*"
    placeholder="hh"
    refCallback={[Function]}
    required={false}
    size="2"
    type="text"
    value="10"
  />
  <span
    className="time-spacer"
  >
    :
  </span>
  <Input
    aria-label="Minutes"
    className="time-input-minute desktop"
    disabled={false}
    isInvalid={false}
    maxLength="2"
    name="terra-time-minute-time-input"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    pattern="\\\\d*"
    placeholder="mm"
    refCallback={[Function]}
    required={false}
    size="2"
    type="text"
    value="45"
  />
</div>
`;

exports[`should render a time input with custom attributes 1`] = `
<div
  className="time-input"
>
  <input
    name="time-input"
    type="hidden"
    value=""
  />
  <Input
    aria-label="Hours"
    className="time-input-hour desktop"
    disabled={false}
    id="terra-time-input"
    isInvalid={false}
    maxLength="2"
    name="terra-time-hour-time-input"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    pattern="\\\\d*"
    placeholder="hh"
    refCallback={[Function]}
    required={false}
    size="2"
    type="text"
    value=""
  />
  <span
    className="time-spacer"
  >
    :
  </span>
  <Input
    aria-label="Minutes"
    className="time-input-minute desktop"
    disabled={false}
    id="terra-time-input"
    isInvalid={false}
    maxLength="2"
    name="terra-time-minute-time-input"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    pattern="\\\\d*"
    placeholder="mm"
    refCallback={[Function]}
    required={false}
    size="2"
    type="text"
    value=""
  />
</div>
`;

exports[`should render a time input with onBlur 1`] = `
<div
  className="time-input"
>
  <input
    name="time-input"
    type="hidden"
    value=""
  />
  <Input
    aria-label="Hours"
    className="time-input-hour desktop"
    disabled={false}
    isInvalid={false}
    maxLength="2"
    name="terra-time-hour-time-input"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    pattern="\\\\d*"
    placeholder="hh"
    refCallback={[Function]}
    required={false}
    size="2"
    type="text"
    value=""
  />
  <span
    className="time-spacer"
  >
    :
  </span>
  <Input
    aria-label="Minutes"
    className="time-input-minute desktop"
    disabled={false}
    isInvalid={false}
    maxLength="2"
    name="terra-time-minute-time-input"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    pattern="\\\\d*"
    placeholder="mm"
    refCallback={[Function]}
    required={false}
    size="2"
    type="text"
    value=""
  />
</div>
`;

exports[`should render a time input with onChange 1`] = `
<div
  className="time-input"
>
  <input
    name="time-input"
    type="hidden"
    value=""
  />
  <Input
    aria-label="Hours"
    className="time-input-hour desktop"
    disabled={false}
    isInvalid={false}
    maxLength="2"
    name="terra-time-hour-time-input"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    pattern="\\\\d*"
    placeholder="hh"
    refCallback={[Function]}
    required={false}
    size="2"
    type="text"
    value=""
  />
  <span
    className="time-spacer"
  >
    :
  </span>
  <Input
    aria-label="Minutes"
    className="time-input-minute desktop"
    disabled={false}
    isInvalid={false}
    maxLength="2"
    name="terra-time-minute-time-input"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    pattern="\\\\d*"
    placeholder="mm"
    refCallback={[Function]}
    required={false}
    size="2"
    type="text"
    value=""
  />
</div>
`;

exports[`should render a time input with onInputFocus 1`] = `
<div
  className="time-input"
>
  <input
    name="time-input"
    type="hidden"
    value=""
  />
  <Input
    aria-label="Hours"
    className="time-input-hour desktop"
    disabled={false}
    isInvalid={false}
    maxLength="2"
    name="terra-time-hour-time-input"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    pattern="\\\\d*"
    placeholder="hh"
    refCallback={[Function]}
    required={false}
    size="2"
    type="text"
    value=""
  />
  <span
    className="time-spacer"
  >
    :
  </span>
  <Input
    aria-label="Minutes"
    className="time-input-minute desktop"
    disabled={false}
    isInvalid={false}
    maxLength="2"
    name="terra-time-minute-time-input"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    pattern="\\\\d*"
    placeholder="mm"
    refCallback={[Function]}
    required={false}
    size="2"
    type="text"
    value=""
  />
</div>
`;

exports[`should set the labels with the correct htmlFor a 24 hour timepicker properly on mobile devices 1`] = `
<div
  class="mobile-time-picker"
>
  <input
    name="time-input"
    type="hidden"
    value=""
  />
  <div
    class="time-input-group"
  >
    <input
      class="form-input time-input-hour"
      id="id-2"
      maxlength="2"
      name="terra-time-hour-time-input"
      pattern="\\\\d*"
      placeholder="hh"
      size="2"
      value=""
    />
    <label
      class="mobile-input-label"
      for="id-2"
    >
      Hours
    </label>
  </div>
  <span
    class="time-spacer"
  >
    :
  </span>
  <div
    class="time-input-group"
  >
    <input
      class="form-input time-input-minute"
      id="id-1"
      maxlength="2"
      name="terra-time-minute-time-input"
      pattern="\\\\d*"
      placeholder="mm"
      size="2"
      value=""
    />
    <label
      class="mobile-input-label"
      for="id-1"
    >
      Minutes
    </label>
  </div>
</div>
`;
